File: web-sdks/dart/lib/src/models/products/part_application.dart
import 'part_application_display.dart';

class PartApplication {
  final int id;
  final bool assets;
  final List<PartApplicationDisplay> displays;

  const PartApplication({
    required this.id,
    required this.assets,
    required this.displays,
  });

  factory PartApplication.fromJson(Map<String, dynamic> json) =>
      PartApplication(
        id: json['id'],
        assets: json['assets'],
        displays: (json['displays'] as List<dynamic>)
            .map(
              (d) => PartApplicationDisplay.fromJson(d as Map<String, dynamic>),
            )
            .toList(),
      );
}

File: web-sdks/javascript/src/models/products/part_application.js
/**
 * Part application models
 */

/**
 * Part application display
 */
export class PartApplicationDisplay {
  constructor({ name, value }) {
    this.name = name;
    this.value = value;
  }

  static fromJson(json) {
    return new PartApplicationDisplay({
      name: json.name,
      value: json.value,
    });
  }
}

/**
 * Part application
 */
export class PartApplication {
  constructor({ id, assets, displays }) {
    this.id = id;
    this.assets = assets;
    this.displays = displays;
  }

  static fromJson(json) {
    return new PartApplication({
      id: json.id,
      assets: json.assets || [],
      displays: (json.display || []).map(d => PartApplicationDisplay.fromJson(d)),
    });
  }
}
